index.js

```diff
@@ -59,47 +91,23 @@ function router (controller, routes, options) {
       delete input.params;
       delete input.query;
 
-      var params = route.match(/:.[^\/]*/g);
-      var url = route;
-
-      if (params) {
-        // If called from a url change, add params and query to input
-        if (input.route.params) {
-          input = params.reduce(function (input, param) {
-            var key = param.substr(1, param.length);
-            input[key] = input.route.params[key];
-            return input;
-          }, input);
-        }
+      var url = getUrl(route, input);
 
-        // Create url based on direct signal input or
-        // params passed from addressbar
-        url = params.reduce(function (url, param) {
-          var key = param.substr(1, param.length);
-          if (!(key in input)) {
-            throw new Error('Cerebral router - The signal "' + routes[route] + '" is bound to "' + route + '" route, but required param "' + key + '" wasn\'t provided.');
-          }
-          return url.replace(param, input[key] || '');
-        }, url);
-
-        // Check resulted url still matches given route
-        var urlMatched = false;
-        var checkRoute = {};
+      // Check resulted url still matches given route
+      var urlMatched = false;
+      var checkRoute = {};
 
-        checkRoute[route] = function () {
-          urlMatched = true;
-        };
+      checkRoute[route] = function () {
+        urlMatched = true;
+      };
 
-        urlMapper(url, checkRoute);
+      urlMapper(url, checkRoute);
 
-        if (!urlMatched) {
-          throw new Error('Cerebral router - Computed url for signal "' + routes[route] +'" can\'t match given route "' + route + '".\n' +
-                          'Check required params provided to signal is not falsy.');
-        }
+      if (!urlMatched) {
+        throw new Error('Cerebral router - Computed url for signal "' + routes[route] +'" can\'t match given route "' + route + '".\n' +
+                        'Check required params provided to signal is not falsy.');
```

Comment:
@christianalfoni seems you have missed that I have changed it in #23 
Falsy values are replaced with empty string.
Use case you have provided seems useful while passing params to signal. I can white list to make `false` appear in url. But again, I found a bit confusing that restored param from url would be a string`'false'` which is `true` in fact. And developer still have to check `input.show !== 'false' || input.show`.

Now my strong suggestion is:
* allow ONLY string values for params used in route definition
* make Boolean values work but log out `console.warn` deprecations